src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/UserManagementServiceTests.cs

```diff
@@ -1017,6 +1017,135 @@ public async Task CancelInvite_InviteNotFound()
+            };
+
+            var publicationRole = new UserPublicationRole
+            {
+                UserId = userId,
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                contentDbContext.Users.Add(internalUser);
+                contentDbContext.UserReleaseInvites.Add(releaseInvite);
+                contentDbContext.UserPublicationInvites.Add(publicationInvite);
+                contentDbContext.UserReleaseRoles.Add(releaseRole);
+                contentDbContext.UserPublicationRoles.Add(publicationRole);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var usersAndRolesDbContext = InMemoryUserAndRolesDbContext(usersAndRolesDbContextId))
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var userManager = MockUserManager();
+                userManager.Setup(mock => mock.DeleteAsync(
+                        It.Is<ApplicationUser>(user => user.Email == email)))
+                    .ReturnsAsync(new IdentityResult());
+
+                var service = SetupUserManagementService(
+                    contentDbContext: contentDbContext,
+                    usersAndRolesDbContext: usersAndRolesDbContext,
+                    userManager: userManager.Object);
+
+                var result = await service.DeleteUser(email);
+                result.AssertRight();
+            }
+
+            await using (var usersAndRolesDbContext = InMemoryUserAndRolesDbContext(usersAndRolesDbContextId))
+            {
+                // Mock checks that ApplicationUser has been removed, so no check required here
+
+                var invites = usersAndRolesDbContext.UserInvites.ToList();
+                Assert.Empty(invites);
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var releaseInvites = contentDbContext.UserReleaseInvites.ToList();
+                Assert.Empty(releaseInvites);
+
+                var publicationInvites = contentDbContext.UserPublicationInvites.ToList();
+                Assert.Empty(publicationInvites);
+
+                var releaseRoles = contentDbContext.UserReleaseRoles.ToList();
+                Assert.Empty(releaseRoles);
+
+                var publicationRoles = contentDbContext.UserPublicationRoles.ToList();
+                Assert.Empty(publicationRoles);
+
+                var dbInternalUserList = contentDbContext.Users.ToList();
+                var dbInternalUser = Assert.Single(dbInternalUserList);
+                Assert.Equal(email, dbInternalUser.Email);
+                Assert.NotNull(dbInternalUser.SoftDeleted);
+                Assert.InRange(DateTime.UtcNow.Subtract(dbInternalUser.SoftDeleted.Value).Milliseconds, 0, 1500);
```

Comment:
We've got a convenience extension `userInvite.Created.AssertUtcNow();` which can be used instead of this.